qiskit/optimization/converters/inequality_to_equality.py

```diff
@@ -71,13 +72,18 @@ def encode(
             QiskitOptimizationError: If an unsupported mode is selected.
             QiskitOptimizationError: If an unsupported sense is specified.
         """
-        self._src = copy.deepcopy(op)
+        self._src = copy.deepcopy(problem)
         self._dst = QuadraticProgram()
+
+        # set a problem name
         if name:
```

Comment:
An empty string, if the user provided it will evaluate False, as does the default value of None. Should this test be `if name is not None` to allow empty string to be set as a problem name. (I see the same test in other convertors of course)